src/contracts/middleware/BLSOperatorStateRetriever.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "./BLSRegistryCoordinatorWithIndices.sol";
+
+import "../interfaces/IStakeRegistry.sol";
+import "../interfaces/IBLSPubkeyRegistry.sol";
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IBLSRegistryCoordinatorWithIndices.sol";
+
+
+contract BLSOperatorStateRetriever {
+    struct Operator {
+        bytes32 operatorId;
+        uint96 stake;
+    }
+
+    struct CheckSignaturesIndices {
+        uint32[] nonSignerQuorumBitmapIndices;
+        uint32[] quorumApkIndices;
+        uint32[] totalStakeIndices;  
+        uint32[][] nonSignerStakeIndices; // nonSignerStakeIndices[quorumNumberIndex][nonSignerIndex]
+    }
+
+    IBLSRegistryCoordinatorWithIndices public registryCoordinator;
+    IStakeRegistry public stakeRegistry;
+    IBLSPubkeyRegistry public blsPubkeyRegistry;
+    IIndexRegistry public indexRegistry;
+
+    constructor(IBLSRegistryCoordinatorWithIndices _registryCoordinator) {
+        registryCoordinator = _registryCoordinator;
+
+        stakeRegistry = _registryCoordinator.stakeRegistry();
+        blsPubkeyRegistry = _registryCoordinator.blsPubkeyRegistry();
+        indexRegistry = _registryCoordinator.indexRegistry();
+    }
+
+    /**
+     * @notice returns the ordered list of operators (id and stake) for each quorum
+     * @param operatorId the id of the operator calling the function
```

Comment:
this address isn't _calling the function_ per se

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/81/commits/48df1fd6538b1ebfede3a4fdb39140d25dbd52d7